src/main/cpp/hw/DragonSparkMax.cpp

```diff
@@ -26,25 +26,37 @@ DragonSparkMax::DragonSparkMax(int id,
-                               double gearRatio) : IDragonMotorController(),
-                                                   m_id(id),
-                                                   m_spark(new CANSparkMax(id, motorType)),
-                                                   // m_controlMode(DRAGON_CONTROL_MODE::PERCENT_OUTPUT),
-                                                   m_outputRotationOffset(0.0),
-                                                   m_gearRatio(gearRatio),
-                                                   m_deviceType(deviceType),
-                                                   m_feedbackType(feedbackType),
-                                                   m_encoder(m_spark->GetEncoder(m_feedbackType)),
-                                                   m_pidController(m_spark->GetPIDController())
+                               rev::SparkLimitSwitch::Type forwardType,
+                               rev::SparkLimitSwitch::Type reverseType,
+                               double gearRatio,
+                               double countsPerDegree,
+                               double countsPerInch) : IDragonMotorController(),
+                                                       m_id(id),
+                                                       m_spark(new CANSparkMax(id, motorType)),
+                                                       m_outputRotationOffset(0.0),
+                                                       m_gearRatio(gearRatio),
+                                                       m_deviceType(deviceType),
+                                                       m_feedbackType(feedbackType),
+                                                       m_forwardType(forwardType),
+                                                       m_reverseType(reverseType),
+                                                       m_encoder(m_spark->GetEncoder(m_feedbackType)),
+                                                       m_pidController(m_spark->GetPIDController()),
+                                                       m_forwardLimitSwitch(m_spark->GetForwardLimitSwitch(m_forwardType)),
+                                                       m_reverseLimitSwitch(m_spark->GetReverseLimitSwitch(m_reverseType)),
+                                                       m_countsPerDegree(countsPerDegree),
+                                                       m_countsPerInch(countsPerInch)
 {
     m_spark->RestoreFactoryDefaults(true);
-
     m_pidController.SetOutputRange(-1.0, 1.0, 0);
     m_pidController.SetOutputRange(-1.0, 1.0, 1);
     m_spark->SetOpenLoopRampRate(0.09); // 0.2 0.25
     m_spark->SetClosedLoopRampRate(0.02);
     m_encoder.SetPosition(0);
     SetRotationOffset(0);
+    m_forwardLimitSwitch.EnableLimitSwitch(false);
+    m_reverseLimitSwitch.EnableLimitSwitch(false);
+    m_forwardType = rev::SparkLimitSwitch::Type::kNormallyOpen;
```

Comment:
Why are the limit switch types being reassigned?

Reply:
Agree, removed